core/store/src/opener.rs

```diff
@@ -147,7 +147,7 @@ fn is_valid_kind_archive(kind: DbKind, archive: bool) -> bool {
     match (kind, archive) {
         (DbKind::Archive, true) => true,
         (DbKind::Cold, true) => true,
-        (DbKind::Hot, true) => true,
+        (DbKind::Hot, _) => true,
```

Comment:
I believe the original reason for the `ensure_kind` check was to fail early in case a node is misconfigured with mismatching hot db, cold db and config. It's not the end of the world if we lose that protection but I'm curious if there is any other good alternative?
* When creating and opening the snapshot, set the right expectations on the kind and temperature? 
* When creating the snapshot overwrite the kind and temperature if needed?

Reply:
I will maybe add refactoring of `opener`x`DbKind` in #10102. At the very least I will add more documentation about default/expected/valid types of DB. The fact that `archive: true` can match two db configurations, and one of those configurations can (from general understanding of our dbs) also match `archive: false` is confusing. 
But I think that as intermediate solution this works. As in this commit can be cherry-picked independently without braking any behaviour, but we still want to make improvements in the future.

Reply:
Agreed, it's been a bit confusing for me as a newcomer to understand what all the DbKind and storage types and archival mean here. Does it maybe make sense to consider waiting till the deprecation of the legacy archival nodes before making more refactoring changes?

Also, thank you @posvyatokum for taking a look at the PR :)